pkg/watermark/store/inmem/kv_store.go

```diff
@@ -79,9 +74,6 @@ func NewKVInMemKVStore(ctx context.Context, pipelineName string, bucketName stri
 		kvEntryCh:    make(chan store.WatermarkKVEntry, 10),
 		log:          logging.FromContext(ctx).With("pipeline", pipelineName).With("bucketName", bucketName),
 	}
-	bucketsLock.Lock()
-	buckets[bucketName] = s
```

Comment:
@chromevoid, why did we have this mapping?

Reply:
I remember correctly it was added because 
1. I thought we will use in mem kv store in real production usages
2. split buckets feature we used to have

Reply:
> I thought we will use in mem kv store for real production use case

why do we need this in real production use cases? How will we use it?

Reply:
> > I thought we will use in mem kv store for real production use case
> 
> why do we need this in real production use cases? How will we use it?

No we don’t need this in real production.